### PR TITLE
Use multi-row inserts for Redshift by default

### PR DESCRIPTION
**Pull request by hariso**
Note: The base branch for this PR is `multi-row-inserts`, which is yet to be merged. Once that one is merged, this one will follow.

**Goal**
Optimal inserts for Redshift, by default. 

**Summary of changes**:
Whenever insert mode is set to `insert` when using the Redshift dialect, the `multi` insert mode will actually be used.

The change is done through the BufferedRecords class, which is responsible for accumulating records in a batch, and flushing it.

### CODE REVIEW

**Comment by hariso** [comment]
@ahmeroxa Please take a look at this PR. Here's some test results: https://github.com/meroxa/aiven-kafka-connect-jdbc/pull/4#issuecomment-932320242

File: src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java [hunk 1]
```diff
@@ -120,21 +122,29 @@ private void prepareStatement() throws SQLException {
...
         return sql;
     }
 
+    private JdbcSinkConfig.InsertMode insertMode() {
+        if (config.insertMode == INSERT && dbDialect instanceof RedshiftDatabaseDialect) {
+            log.debug("Multi-row inserts will be used, because {} supports it.", dbDialect.name());
+            return MULTI;
+        }
+        return config.insertMode;
+    }
```
Comment (ahmeroxa) [inline src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java:146]: I don't think we should force `multi` if the config has been explicitly set to `insert`.

Comment (hariso) [reply src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java:146]: I actually agree with you, but the insert mode is assigned a default value when a JdbcSinkConfig instance is created. Even if no insert mode has been specified, there will be config.insertMode == INSERT.

Comment (hariso) [reply src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java:146]: Actually, I might have found the original input, I'll double check this. Thanks for checking this out!

Comment (hariso) [reply src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java]: Done in https://github.com/meroxa/aiven-kafka-connect-jdbc/pull/5/commits/292c3ae4c45d911ef68ff7e544a58f9045e46e7e

**Comment by hariso** [comment]
@ahmeroxa Please re-review, I've addressed your concern about overriding explicitly set insert modes.

**Review by ahmeroxa — APPROVED** [review]
LGTM 👍

**Pull request merged**